src/backend/cuda/types.hpp

```diff
@@ -149,16 +149,13 @@ struct kernel_type<common::half> {
     using data = common::half;
 
 #ifdef __CUDA_ARCH__
-
     // These are the types within a kernel
-
-#if __CUDA_ARCH__ > 530 && __CUDA_ARCH__ != 610
+#if __CUDA_ARCH__ >= 530 && __CUDA_ARCH__ != 610
     using compute = __half;
 #else
     using compute = float;
 #endif
 #else
-
```

Comment:
`__CUDA_ARCH__` is defined for nvrtc compilation also. The only time `#else` of the `#ifdef __CUDA_ARCH__` is taken is when this header is included by `.cpp` in CUDA backend.

The following(not in this PR's diff)
```
#if defined(NVCC) || defined(__CUDACC_RTC__)
    using native  = __half;
#else
    using native = common::half;
#endif
```
can be just
```
#if defined(NVCC)
    using native  = __half;
#else
    using native = common::half;
#endif
```
Don't you think so ?

Reply:
This is an obsolete check and doesn't hurt the builds at the moment. Deferring to fix in later PR.